### PR TITLE
Desktop support documentation

### PR DESCRIPTION
**Pull request by ImmanuelBaskaran**
Added documentation regarding the new style injection system

### CODE REVIEW

File: packages/styles/stories/introduction.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import { Meta } from "@storybook/addon-docs";
+
+<Meta title="Styles/About the Style injection system" />
+
+# About the Salt Theme
```
Comment (origami-z) [inline packages/styles/stories/introduction.stories.mdx:5]: update?

**Comment by libertymayc** [comment]
lgtm
regarding IA, @pseys may have a recommendation on where 'styles' should appear in the sidebar

File: packages/styles/stories/introduction.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+## Component style injection
+
+In order to support uses of Salt that utilize multiple external windows such as desktop applications. We offer two packages that contain hooks to facilitate this: `@salt-ds/window` and `@salt-ds/styles`.
```
Comment (origami-z) [inline packages/styles/stories/introduction.stories.mdx:9]: `desktop applications` is not a well understood concept among web devs, could illustrate more on what it is and what exact problem to be addressed.

File: packages/styles/stories/introduction.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import { Meta } from "@storybook/addon-docs";
+
+<Meta title="Styles/About the Style injection system" />
```
Comment (origami-z) [inline packages/styles/stories/introduction.stories.mdx:3]: Probably need some background information around what is "style injection", what's common solution out there, pro/cons, etc.

File: packages/styles/stories/introduction.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    window: targetWindow,
+  });
+
+  return <svg className="saltExample-svg">{props.children}</svg>;
+};
+```
+
+- All Salt component CSS must be imported using the default export, the imported CSS should be injected by calling the `useComponentCssInjection` hook.
+- The current window the component is being rendered into is retrieved using the `useWindow` hook.
+- Only one component should be stored per typescript jsx file, however a component may import and inject multiple css files.
```
Comment (origami-z) [inline packages/styles/stories/introduction.stories.mdx:33]: not sure what exactly are we talking about here. Does that mean the system will break if having multiple components in a single file?

File: packages/styles/stories/introduction.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+In the above example, `useComponentCssInjection` is called with an ID which is used to keep track of if component CSS has already been injected before,
+the CSS string itself and the window to inject the css into. Internally, the hook uses a Map of Map objects to store if a CSS string has been injected into a given window.
```
Comment (origami-z) [inline packages/styles/stories/introduction.stories.mdx:36]: May need some rethink on `id` from meetings feedback (e.g. "display name" of React contex). 

By reading this, it feels that `id` is actually quite important concept where clash of id may cause un-intended outcome.

File: packages/styles/stories/introduction.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+the element where `useComponentCssInjection` inserts the styles can be controlled using an `InsertionPointProvider`. The hook will then insert the given CSS string before the provided insertion point
+
+```tsx
+import { useRef } from "react";
+import { Button } from "@salt-ds/core";
+import { InsertionPointProvider } from "@salt-ds/styles";
+
+const insertionPointRef = useRef<HTMLElement>();
+
+<InsertionPointProvider insertionPoint={insertionPointRef.current}>
```
Comment (origami-z) [inline packages/styles/stories/introduction.stories.mdx:49]: Could have more elaboration on why is this important, maybe with some real life example.

Comment (ImmanuelBaskaran) [reply packages/styles/stories/introduction.stories.mdx:59]: I've only got one example for this at the moment, I'm open to any further examples.

File: packages/styles/stories/introduction.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+Developers may want to control where the CSS is injected into the document, such as if a component library's styles are injected at the bottom of the head element
+and overriding an application's styles or injecting styles into a shadow dom.
+
+the location where `useComponentCssInjection` inserts the styles can be controlled using an `InsertionPointProvider`. The hook will then insert the given CSS string before the provided insertion point
+
+```tsx
+import { Button } from "@salt-ds/core";
+import { InsertionPointProvider } from "@salt-ds/styles";
+
+<InsertionPointProvider insertionPoint={document.head.lastElementChild}>
```
Comment (origami-z) [inline packages/styles/stories/introduction.stories.mdx:58]: Can give more detail on this line of code

**Comment by origami-z** [comment]
I'm happy with the doc. Need confirmation on order of sidebar.

May worth to ask others who haven't worked on desktop to review

**Pull request merged**